internal/internal_worker_base.go

```diff
@@ -209,6 +215,9 @@ func (bw *baseWorker) Start() {
 	bw.shutdownWG.Add(1)
 	go bw.runTaskDispatcher()
 
+	bw.shutdownWG.Add(1)
+	bw.EmitHardwareUsage()
```

Comment:
Not sure if that's the best practice of using waitgroup and go routine (since the go routine is started inside this function) but the test is not reporting and goleak.